### ISSUE TITLE
Computer Vision Model

### ISSUE THREAD

**Issue by timrleahy** [260223-DF-AI/TimLeeAIProject#1]
The declaration for the AI, including the layers, forward pass, and training methods will go in model.py.

Must satisfy these P3 requirements:
- Use a custom nn.Module or a fine-tuned ResNet backbone.
- Implement Mixed Precision (AMP) and Early Stopping based on a validation set.
